sink/datadog.go

```diff
@@ -276,9 +278,31 @@ func (s *Datadog) Send(ctx context.Context, m *blip.Metrics) error {
 
 			// Convert Blip metric type to Datadog metric type
 			switch metrics[i].Type {
-			case blip.COUNTER:
+			case blip.CUMULATIVE_COUNTER, blip.DELTA_COUNTER:
+				metricVal := metrics[i].Value
+
+				if metrics[i].Type == blip.CUMULATIVE_COUNTER {
+					// if the value is not specifically marked as DELTA, calculate delta
+
+					metricID := s.uniqIdentifierForMetric(name, metrics[i].Group)
```

Comment:
"Naming things" :-) Abbreviating "unique" to "uniq" is not standard because it's not much of an abbreviation when it's only off by 2: "ue". Plus abbreviating that but spelling out "identifier" (10 chars) instead of using "id" is weird (why save 2 but not 8?). And identifiers are unique by definition, so the two words together are redundant. And since Go likes short names "forMetric" isn't adding any value given the args and the value to which the return value is assigned.

Instead: `metricId := s.metricId(name,  metrics[i].Group))`.  And below:

```go
// metricId returns the metric name concatenated with sorted group keys.
// For example, if the metric name is "foo" and the group keys are "a" and "b",
// it returns "fooab". This is used to calculate delta counter values in Collect.
func (s *Datadog) metricId(name string, groups map[string]string) string {
```

Reply:
Refactored.